athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/storage/StorageMetadata.java

```diff
@@ -0,0 +1,315 @@
+                for (Blob blob : blobPage.iterateAll()) {
+                    String blobName = blob.getName();
+                    String folderPath = blobName.startsWith(path)
+                            ? blobName.replace(path, "")
+                            : blobName;
+                    // remove the front-slash, because, the expression generated without it
+                    if (folderPath.startsWith("/")) {
+                        folderPath = folderPath.substring(1);
+                    }
+                    LOGGER.info("Examining folder {} against regex {}", folderPath, folderRegEx);
+                    if (folderRegExPattern.matcher(folderPath).matches()) {
+                        LOGGER.info("Examining folder {} against regex {} matches", folderPath, folderRegEx);
+                        if (!canIncludePath(folderPath, expressions)) {
+                            LOGGER.info("Folder {} has NOT been selected against the expression", folderPath);
+                            continue;
+                        }
+                        LOGGER.info("Folder {} has been selected against the expression", folderPath);
+                        Map<String, String> tableParameters = table.getParameters();
+                        List<PartitionColumnData> partitions = PartitionUtil.getStoragePartitions(tableParameters.get(PARTITION_PATTERN_KEY), folderPath,
+                                folderRegEx, table.getPartitionKeys());
+                        if (!partitions.isEmpty()) {
+                            partitionFolders.add(partitions);
+                        }
+                        else {
+                            LOGGER.info("No partitions found for the folder {}", blob.getName());
+                        }
+                    }
+                }
+            }
+        }
+        else {
+            LOGGER.info("Table {}.{} not found", tableInfo.getSchemaName(), tableInfo.getTableName());
+        }
+        return partitionFolders;
+    }
+
+    /**
+     * Retrieves a first of file has non-zero size
+     *
+     * @param bucket Name of the bucket
+     * @param prefix Prefix (aka, folder in Storage service) of the bucket from where this method with retrieve files
+     * @return A single file name under the prefix
+     */
+    protected Optional<String> getStorageFiles(String bucket, String prefix)
+    {
+        LOGGER.info("Listing nested files for prefix {} under the bucket {}", prefix, bucket);
+        Page<Blob> blobPage = storage.list(bucket, prefix(prefix.substring(1)));
+        for (Blob blob : blobPage.iterateAll()) {
+            // check whether it is file, It may return folder also
+            if (blob.getSize() > 0) {
+               return Optional.of(blob.getName());
+            }
+        }
+        return Optional.empty();
+    }
+
+    // helpers
+    public Schema getFileSchema(String bucketName, String path, FileFormat format, BufferAllocator allocator)
```

Comment:
helpers  should not be public. I see we're using it in a test class, so make it protected and mark it `@VisibleForTesting` .

Reply:
resolved